lib/lure/command/updateDependencies.go

```diff
@@ -0,0 +1,217 @@
+package command
+
+import (
+	"fmt"
+	"os"
+	"reflect"
+	"strings"
+
+	"github.com/coveooss/lure/lib/lure"
+	"github.com/coveooss/lure/lib/lure/log"
+	"github.com/coveooss/lure/lib/lure/project"
+	p "github.com/coveooss/lure/lib/lure/provider"
+	"github.com/coveooss/lure/lib/lure/versionManager"
+	"github.com/coveooss/lure/lib/lure/versionManager/mvn"
+	"github.com/coveooss/lure/lib/lure/versionManager/npm"
+
+	"github.com/vsekhar/govtil/guid"
+)
+
+// This part interesting
+// https://github.com/golang/go/blob/1441f76938bf61a2c8c2ed1a65082ddde0319633/src/cmd/go/vcs.go
+
+func appendIfMissing(modules []versionManager.ModuleVersion, modulesToAdd []versionManager.ModuleVersion) []versionManager.ModuleVersion {
+	for _, moduleToAdd := range modulesToAdd {
+		exist := false
+		for _, module := range modules {
+			if (moduleToAdd.Name != "" && module.Name == moduleToAdd.Name) || reflect.DeepEqual(module, moduleToAdd) {
+				exist = true
+				break
+			}
+		}
+		if !exist {
+			modules = append(modules, moduleToAdd)
+		}
+	}
+
+	return modules
+}
+
+func CheckForUpdatesJobCommand(project project.Project, sourceControl sourceControl, provider provider, args map[string]string) error {
+	return checkForUpdatesJob(project, sourceControl, provider, args["commitMessage"])
+}
+
+func checkForUpdatesJob(project project.Project, sourceControl sourceControl, provider provider, commitMessage string) error {
+	log.Logger.Infof("switching to default branch: %s", project.DefaultBranch)
+	if _, err := sourceControl.Update(project.DefaultBranch); err != nil {
+		return fmt.Errorf("Error: \"Could not switch to branch %s\" %s", project.DefaultBranch, err)
+	}
+
+	modulesToUpdate := make([]versionManager.ModuleVersion, 0, 0)
```

Comment:
The creation of `moduleToUpdate` could be put into its own `getModulesToUpdate()` function.